keyrings/alt/file_base.py

```diff
@@ -43,16 +75,20 @@ class Keyring(FileBacked, KeyringBackend):
     """
 
     @abc.abstractmethod
-    def encrypt(self, password):
+    def encrypt(self, password, assoc = None):
         """
         Given a password (byte string), return an encrypted byte string.
+
+        assoc might provide associated data (typically: service and username)
```

Comment:
The docstring should describe what assoc is (byte string) and what it does (what's its purpose).